### PR TITLE
DRYD-827: UI Changes for UoC Record

### PR DESCRIPTION
**Pull request by cesarvh**

### CODE REVIEW

**Comment by cesarvh** [comment]
@ray-lee  Not sure why the coveralls is decreasing. I cherry-picked this right onto the tip of the master branch on the collectionspace repo. Let me know if there's anything I need to do to fix this. Same with #117  and #115

File: src/plugins/recordTypes/uoc/idGenerators.js [hunk 1]
```diff
@@ -10,4 +10,13 @@ export default {
...
   },
+  proj: {
+    csid: 'ddba36e0-0534-495d-84ea-8b8b65a1f79f',
+    messages: defineMessages({
+      type: {
+        id: 'idGenerator.proj.type',
+        defaultMessage: 'Use of Collections Project',
+      },
+    }),
+  },
```
Comment (ray-lee) [inline src/plugins/recordTypes/uoc/idGenerators.js]: This needs to be configured in application layer XML too.

File: src/plugins/recordTypes/uoc/fields.js [hunk 2]
```diff
@@ -60,6 +64,38 @@ export default (configContext) => {
...
+            messages: defineMessages({
+              name: {
+                id: 'field.uoc_common.projectId.name',
+                defaultMessage: 'Project ID',
+              },
+            }),
+            view: {
+              type: TermPickerInput,
+              props: {
+                source: 'uocprojectid',
```
Comment (ray-lee) [inline src/plugins/recordTypes/uoc/fields.js]: All of these new term lists need to be initialized in app layer config.

File: src/plugins/recordTypes/uoc/fields.js [hunk 3]
```diff
@@ -98,68 +134,194 @@ export default (configContext) => {
...
-              type: DateInput,
+              type: CompoundInput,
+            },
+          },
+          useDateGroup: {
+            [config]: {
+              messages: defineMessages({
+                name: {
+                  id: 'field.uoc_common.useDateGroup.name',
+                  defaultMessage: 'Start/single date', // FIX ME
```
[the remark this answers is not in the document]
Comment (remillet) [reply src/plugins/recordTypes/uoc/fields.js]: Ray, you should check with Megan about this label.  Maybe the current label is trying to deal with cases when there is a "Start" date and also when there is just a one-day/single date?

[the remark this answers is not in the document]
Comment (remillet) [reply src/plugins/recordTypes/uoc/fields.js]: Ray, after checking with Megan, feel free to change this label to whatever you guys think is best.

[the remark this answers is not in the document]
Comment (ray-lee) [reply src/plugins/recordTypes/uoc/fields.js]: @remillet The issue is that you made it repeating. It was fine as a label before.

**Pull request merged**